### PR TITLE
Docker setup

### LINKED ISSUE #9: [DEVOPS] Initial setup
## ⚙️ DevOps

Initial setup of the project infrastructure with the MERN stack, and set the foundation for the rest of the project development life cycle

### Motivation

After this is done the development team will be able to start working on both the backend and frontend of the app

### Implementation

It should include:

* express-node server with typescript and it's file structure
* react frontend app with typescript, tailwind, and it's file structure
* scripts for building and running locally, and with docker
* relevant readme files (how to run, contribution standards, etc)

### PR DESCRIPTION
**Pull request by MelnikAndres**
## Motivation

Closes #9 

## Summary of changes
Better Docker setup using compose. Separated containers for front-end and back-end. Nginx implementation as inverse proxy. App port is now 3050.
Tiny readme update

## Checklist
- [ ] Tested the changes locally.
- [ ] Reviewed the changes on GitHub, line by line.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

File: truco-back/src/app.ts [hunk 1]
```diff
@@ -3,10 +3,7 @@ import path from "path";
...
 const app = express();
 
-app.use(express.static(path.join(__dirname, 'public')));
-
-app.use("/", (req, res) => {
-    res.sendFile(path.join(__dirname, '/public/index.html'));
+app.get("/api", (req, res) => {
+    res.send("Hello World");
 })
-
```
Comment (maxogod) [inline truco-back/src/app.ts:11]: Without this the app is no longer serving the static files from the backend which means the app will be separated into two different servers, one for the frontend and one for the backend. That's fine during development as we will use CORS, but we should not delete it because in the PROD version of the app there will only be one server, and without this lines it won't be possible

Comment (maxogod) [reply truco-back/src/app.ts:11]: just to make it clear, the api WILL be at xxxxxx/api but when a user enters xxxxxx/ they should see the frontend homepage not a `Cannot GET /` message

image
image

File: Dockerfile
Comment (maxogod) [inline Dockerfile:1]: why was this deleted? we still need a way to build the entire app into one singular container for when the app is finished

**Review by maxogod — CHANGES_REQUESTED** [review]

File: truco-back/src/app.ts (hunk 1 above)
Comment (MelnikAndres) [reply truco-back/src/app.ts:11]: With this implementation, nginx is used as an inverse proxy. If you go to localhost:3050/api you will find the backend
if you go to localhost:3050, you will find the frontend.

File: Dockerfile
Comment (MelnikAndres) [reply Dockerfile:1]: That Dockerfile is not the right way to do it.
